### PR TITLE
feat: Process updates on peer

### PR DESCRIPTION
**Pull request by pablitar**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -32,6 +32,7 @@
       "decentraland-katalyst-contracts/*": ["contracts/*"],
       "decentraland-katalyst-commons/*": ["commons/servers/*"],
       "decentraland-catalyst-utils/*": ["commons/utils/*"],
+      "comms-protocol/*": ["comms/protocol/*"],
```
Comment (nchamo) [inline tsconfig.json:35]: YYYEES! Haha

File: comms/protocol/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    name = "comms-protocol",
+    srcs = glob(["*.ts"]),
+    module_name = "comms-protocol",
+    tsconfig = "//:tsconfig.json",
+    deps = [
+        "//commons/utils",
+    ],
+)
+
+# ts_library(
```
Comment (nchamo) [inline comms/protocol/BUILD.bazel:16]: Should we delete this?

File: commons/utils/Positions.ts [hunk 3]
```diff
@@ -1,19 +1,8 @@
...
-export type Position = Position2D | Position3D
-
-export function isPosition3D(position: any): position is Position3D {
-  return position instanceof Array && position.length === 3
-}
-
-export function isPosition2D(position: any): position is Position2D {
-  return position instanceof Array && position.length === 2
-}
+export type Position3D = [number, number, number]
```
Comment (nchamo) [inline commons/utils/Positions.ts:5]: Best change in the PR haha

File: comms/peer/test/Peer.integration.spec.ts [hunk 4]
```diff
@@ -34,7 +28,7 @@ type PositionedPeer = {
 describe('Peer Integration Test', function () {
   const DEFAULT_LIGHTHOUSE = 'http://notimportant:8888'
 
-  let lighthouses: Record<string, LighthouseState>
+  let lighthouses: Record<string, Record<string, MinPeerData & { islandId?: string }>>
```
Comment (nchamo) [inline comms/peer/test/Peer.integration.spec.ts:31]: Wow, a lot of work on this file!

Thank you for doing it! ❤️

File: comms/peer/src/Peer.ts [hunk 5]
```diff
@@ -368,67 +340,59 @@ export class Peer implements IPeer {
...
 
-  async setLayer(layer: string): Promise<void> {
-    if (this.disposed) return
-    const id = this.peerIdOrFail()
-    const { json } = await this.httpClient.fetch(`/layers/${layer}`, {
-      method: 'PUT',
-      // userId and peerId are deprecated but we leave them here for compatibility. When all lighthouses are safely updated they should be removed
-      bodyObject: { id, userId: id, peerId: id, protocolVersion: PROTOCOL_VERSION }
-    })
+  // async setLayer(layer: string): Promise<void> {
```
Comment (nchamo) [inline comms/peer/src/Peer.ts:343]: Should we remove this?

**Review by nchamo — APPROVED** [review]
Great work Pablo!!!

File: comms/protocol/BUILD.bazel (hunk 2 above)
Comment (pablitar) [reply comms/protocol/BUILD.bazel:16]: Totally. Will do

File: comms/peer/src/Peer.ts (hunk 5 above)
Comment (pablitar) [reply comms/peer/src/Peer.ts:343]: Totally.

**Pull request merged**